staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -232,20 +232,11 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	proxy := httputil.NewSingleHostReverseProxy(&url.URL{Scheme: h.Location.Scheme, Host: h.Location.Host})
 	proxy.Transport = h.Transport
 	proxy.FlushInterval = h.FlushInterval
-	proxy.ErrorLog = log.New(noSuppressPanicError{}, "", log.LstdFlags)
-	proxy.ServeHTTP(w, newReq)
-}
-
-type noSuppressPanicError struct{}
 
-func (noSuppressPanicError) Write(p []byte) (n int, err error) {
-	// skip "suppressing panic for copyResponse error in test; copy error" error message
-	// that ends up in CI tests on each kube-apiserver termination as noise and
-	// everybody thinks this is fatal.
-	if strings.Contains(string(p), "suppressing panic") {
-		return len(p), nil
-	}
-	return os.Stderr.Write(p)
+	defer func() {
+		recover() // ignore panics inside of the reverse proxy on unclean disconnects
```

Comment:
`klog.V(4)` so we have some shot at debugging "weird" issues if they come up?

Reply:
why isn't our existing WithPanicRecovery handling this already?

Reply:
also, if we're using this as an expected control mechanism, I'd expect catching/comparing with http.ErrAbortHandler, and propagating the panic otherwise (preserving the original stack as we do in other places we catch/propagate)

Reply:
it handles it, but prints it, giving you hundred of panics in the logs

Reply:
ah. looks like if we're going to inject ourselves into http server panic handling, we have to honor `http.ErrAbortHandler`:

> ErrAbortHandler is a sentinel panic value to abort a handler. While any panic from ServeHTTP aborts the response to the client, panicking with ErrAbortHandler also suppresses logging of a stack trace to the server's error log.